### ISSUE TITLE
VJS 5: Tech Unit Test Specs

### ISSUE THREAD

**Issue by eXon** [videojs/video.js#2004]
When you want to build a tech, you are completely left in the dark. You don't know what you are doing. The only hope you have is to look at how the HTML5 and Flash techs are built and do some trial and errors to make your own work. Just knowing if you are handling every possible events you could is hard.

There is something we can do about it. What about a set of unit test specs that you could run any tech and spit out what is broken?

- We could write tests using something readable like Cucumber or straight unit tests
- We could build a NullTech and keep it up-to-date with the tests
(easier to understand when you build something from scratch)
- Then, anyone using NPM could get those tests and run it on his own tech
(that way, he knows his tech is working fine and when we do regression, they immediately know something is broken)

**Comment by heff** [comment]
Yeah, I think this would be great. We talked about this (or at least a similar idea) over a year ago in #835. 835 is more of a template while this issue is more of a TDD approach. But both I think could be combined in a yeoman generator. I think we should get through some of the other [Tech 2.0](https://github.com/videojs/video.js/milestones/Techs%202.0) architecture pieces first because some core things will change, but I'd love to see this done if you'd be interested in working on it.

**Comment by eXon** [comment]
I also have a proposal for Tech 2.0 and how it should interact with the player. I think it will blend well with what you had in mind on #1485

I'll send a first draft in the upcoming weeks.
